server/src/services/check.service.ts

```diff
@@ -0,0 +1,96 @@
+import { getRepository } from 'typeorm';
+import { CourseTask, Student, Task, TaskSolutionResult, User } from '../models';
+
+const LOW_ERROR_RATE = 0.9;
+const HIGH_ERROR_RATE = 1.1;
+const TASK_REVIEW_COUNT = 4;
+const MIN_LENGTH_MESSAGE = 70;
+
+/* Get checkers who passed max score for everyone and maybe didn't review task*/
+export async function getCheckersWithMaxScore(taskId: number) {
+  const data = await getRepository(TaskSolutionResult)
```

Comment:
Really huge query...
Did you test it on some "big" data? It is work quite fast?

Reply:
I didn't test( because i didn't have big data
I asked for snapshot from prod app, but i didn't get

Reply:
I think you can use some tool for generating random data like this (https://www.mockaroo.com/).
It quite configurable.
Just fill tables that you need with some random data (I think max of 10000 students will be enough).
And then run it locally to check (at least use console.time).

If you need some help with generation random data - we can make some short call, just let me know

Reply:
I wouldn't refuse a call, because in this table pairs(checker, student) must be unique

I have made the query shorter